src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_TryCatch.cs

```diff
@@ -116,5 +120,638 @@ void M(int i)
+          Variables: Local_1: System.Int32 i
+          Initializer: ILiteralExpression (OperationKind.LiteralExpression, Type: System.Int32, Constant: 0) (Syntax: '0')
+  Catch clauses(1):
+      ICatchClause (Exception type: System.Exception) (OperationKind.CatchClause) (Syntax: 'catch (Exce ... }')
+        ExceptionDeclarationOrExpression: null
+        Filter: null
+        Handler: IBlockStatement (0 statements) (OperationKind.BlockStatement) (Syntax: '{ ... }')
+  Finally: null
+";
+            var expectedDiagnostics = new DiagnosticDescription[] {
+                // CS0219: The variable 'i' is assigned but its value is never used
+                //             int i = 0;
+                Diagnostic(ErrorCode.WRN_UnreferencedVarAssg, "i").WithArguments("i").WithLocation(10, 17)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<TryStatementSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void TryCatch_CatchClauseWithLocalDeclaration()
+        {
+            string source = @"
+using System;
+
+class C
+{
+    static void M(string s)
+    {
+        /*<bind>*/try
+        {
+        }
+        catch (Exception)
+        {
+            int i = 0;
+        }/*</bind>*/
+    }
+}
+";
+            string expectedOperationTree = @"
+ITryStatement (OperationKind.TryStatement) (Syntax: 'try ... }')
+  Body: IBlockStatement (0 statements) (OperationKind.BlockStatement) (Syntax: '{ ... }')
+  Catch clauses(1):
+      ICatchClause (Exception type: System.Exception) (OperationKind.CatchClause) (Syntax: 'catch (Exce ... }')
+        ExceptionDeclarationOrExpression: null
+        Filter: null
+        Handler: IBlockStatement (1 statements, 1 locals) (OperationKind.BlockStatement) (Syntax: '{ ... }')
+            Locals: Local_1: System.Int32 i
+            IVariableDeclarationStatement (1 declarations) (OperationKind.VariableDeclarationStatement) (Syntax: 'int i = 0;')
+              IVariableDeclaration (1 variables) (OperationKind.VariableDeclaration) (Syntax: 'i = 0')
+                Variables: Local_1: System.Int32 i
+                Initializer: ILiteralExpression (OperationKind.LiteralExpression, Type: System.Int32, Constant: 0) (Syntax: '0')
+  Finally: null
+";
+            var expectedDiagnostics = new DiagnosticDescription[] {
+                // CS0219: The variable 'i' is assigned but its value is never used
+                //             int i = 0;
+                Diagnostic(ErrorCode.WRN_UnreferencedVarAssg, "i").WithArguments("i").WithLocation(13, 17)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<TryStatementSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void TryCatch_CatchFilterWithLocalDeclaration()
```

Comment:
Locals are currently not exposed in the operation tree. https://github.com/dotnet/roslyn/issues/22301 tracks adding this API.